### PR TITLE
Add unit test for auth removal

### PR DESCRIPTION
**Pull request by moelsayed**
**Description**

Changes proposed in this pull request:

- Add unit test for auth removal
- Remove dead code
- ...

**Related issue(s)**

https://github.com/kyma-project/kyma/issues/14593

### CODE REVIEW

File: components/function-controller/internal/webhook/conversion_webhook_test.go [hunk 1]
```diff
@@ -380,6 +380,82 @@ func TestConvertingWebhook_convertFunction(t *testing.T) {
...
+	tests := []struct {
+		name        string
+		src         runtime.Object
+		repo        serverlessv1alpha1.GitRepository
+		wantDst     runtime.Object
+		wantVersion string
+		wantErr     bool
+	}{
+		{
+			name: "v1alpha1 to v1alpha2 Git function - auth removed",
```
Comment (dbadura) [inline components/function-controller/internal/webhook/conversion_webhook_test.go:407]:
```suggestion
			name: "v1alpha1 to v1alpha2 Git function without auth",
```

**Review by dbadura — COMMENTED** [review]
Why do we need seperate test for this?
Can we just add new test case?

**Comment by moelsayed** [comment]
@dbadura it will just complicate the other test cases and the test body. I have already started with that as you can probably tell by the leftover code :smile:

**Review by dbadura — APPROVED** [review]

**Pull request merged**